source/LocationFilter.cpp

```diff
@@ -504,6 +505,150 @@ const Planet *LocationFilter::PickPlanet(const System *origin, bool hasClearance
 
 
 
+// Members of the NearOrDistanceFilter class.
+
+void LocationFilter::NearOrDistanceFilter::Load(const DataNode &node, int valueIndex)
+{
+	if(node.Size() > valueIndex)
+		LoadElement(node, valueIndex);
+	else if(node.HasChildren())
+		for(const auto &child : node)
+			LoadElement(child, 0);
+}
+
+
+
+LocationFilter::NearOrDistanceFilter::NearElement::NearElement(
+			const LocationFilter::NearOrDistanceFilter::DistanceElement &other)
+	: center(nullptr), minDistance(other.minDistance), maxDistance(other.maxDistance),
+			distanceOptions(other.distanceOptions)
+{
+}
+
+
+
+// Members of the DistanceFilter class.
+
+// At least one of the elements must match.
+bool LocationFilter::DistanceFilter::Matches(const System *system, const System *origin) const
+{
+	for(const auto &it : elements)
+		if(it.maxDistance >= 0 && Distance(origin, system, it.maxDistance, it.distanceOptions) >= it.minDistance)
+			return true;
+	return false;
+}
+
+
+
+const list<LocationFilter::NearOrDistanceFilter::DistanceElement> &LocationFilter::DistanceFilter::Elements() const
+{
+	return elements;
+}
+
+
+
+void LocationFilter::DistanceFilter::LoadElement(const DataNode &node, int index)
+{
+	if(node.Size() < index + 1)
+		return;
+
+	elements.emplace_back();
+	DistanceElement &it = elements.back();
```

Comment:
since C++17 emplace returns a reference to the element
```suggestion
	DistanceElement &it = elements.emplace_back();
```